examples/api/test/widgets/system_context_menu/system_context_menu.1_test.dart

```diff
@@ -0,0 +1,120 @@
+            final MediaQueryData mediaQueryData = MediaQuery.of(context);
+            return MediaQuery(
+              data: mediaQueryData.copyWith(
+                supportsShowingSystemContextMenu: defaultTargetPlatform == TargetPlatform.iOS,
+              ),
+              child: const example.SystemContextMenuExampleApp(),
+            );
+          },
+        ),
+      );
+
+      final Finder textFinder = find.byType(EditableText);
+      await tester.longPress(textFinder);
+      tester.state<EditableTextState>(textFinder).showToolbar();
+      await tester.pumpAndSettle();
+
+      if (defaultTargetPlatform == TargetPlatform.iOS) {
+        expect(find.byType(SystemContextMenu), findsOneWidget);
+        expect(itemsReceived.length, greaterThanOrEqualTo(3));
+        
+        final List<Map<String, dynamic>> customItems = itemsReceived
+            .where((Map<String, dynamic> item) => item['type'] == 'custom')
+            .toList();
+        
+        expect(customItems.length, 3);
+        expect(customItems[0]['title'], 'Clear Text');
+        expect(customItems[1]['title'], 'Add Heart');
+        expect(customItems[2]['title'], 'Uppercase');
+      } else {
+        expect(find.byType(AdaptiveTextSelectionToolbar), findsOneWidget);
+        expect(find.byType(SystemContextMenu), findsNothing);
+      }
+    },
+    variant: TargetPlatformVariant.all(),
+    skip: kIsWeb, // [intended]
+  );
+
+  testWidgets(
+    'custom menu actions work correctly',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        Builder(
+          builder: (BuildContext context) {
+            final MediaQueryData mediaQueryData = MediaQuery.of(context);
+            return MediaQuery(
+              data: mediaQueryData.copyWith(
+                supportsShowingSystemContextMenu: true,
+              ),
+              child: const example.SystemContextMenuExampleApp(),
+            );
+          },
+        ),
+      );
+
+      final TextEditingController controller = tester
+          .widget<TextField>(find.byType(TextField))
+          .controller!;
+
+      expect(controller.text, 'Long press to see custom menu items');
+
+      final Finder textFinder = find.byType(EditableText);
+      await tester.longPress(textFinder);
+      tester.state<EditableTextState>(textFinder).showToolbar();
+      await tester.pumpAndSettle();
+
+      final SystemContextMenu contextMenu = tester.widget<SystemContextMenu>(
+        find.byType(SystemContextMenu),
+      );
+      final List<IOSSystemContextMenuItem> items = contextMenu.items;
+      final IOSSystemContextMenuItemCustom clearItem = items
+          .whereType<IOSSystemContextMenuItemCustom>()
+          .firstWhere((IOSSystemContextMenuItemCustom item) => item.title == 'Clear Text');
+
+      clearItem.onPressed();
+      await tester.pumpAndSettle();
+
+      expect(controller.text, '');
+      expect(find.text('Text cleared'), findsOneWidget);
```

Comment:
Can you also check whether the SystemContextMenu is still visible or not? I assume the answer is no, it hides itself? If not the we should show how to do that in the example, but I think it does hide.